mkosi.build

```diff
@@ -112,10 +112,12 @@ if [ -n "$IMAGE_VERSION" ] ; then
 fi
 
 # Manually update the boot loader from the one we just built
-mkdir -p "$DESTDIR"/boot/efi/EFI/systemd "$DESTDIR"/boot/efi/EFI/BOOT
-cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR"/boot/efi/EFI/systemd/systemd-bootx64.efi
-cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR"/boot/efi/EFI/BOOT/bootx64.efi
-
-mkdir -p "$DESTDIR"/efi/EFI/systemd "$DESTDIR"/efi/EFI/BOOT
-cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR"/efi/EFI/systemd/systemd-bootx64.efi
-cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR"/efi/EFI/BOOT/bootx64.efi
+if [ -f "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi ]; then
```

Comment:
Wait, this is hardcoding efi arch. Even if mkosi is only x86_x64, this should be using a glob…

Reply:
yeah, writing this via a glob should be easy

Reply:
I'm not a bash expert. How would you write `cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR"/efi/EFI/BOOT/bootx64.efi` using globs?

Reply:
uh, eh, oh, ah. @keszybz do you have an idea?

Reply:
Actually, maybe we can do away with this alltogether by writing just writing a postinst script and running bootctl install.

Reply:
This ought to do it:
```
find "$DESTDIR/usr/lib/systemd/boot/efi/" -name "systemd-boot*.efi" | while read -r sdboot; do
    sdboot="$(basename "$sdboot")"
    fallback="${sdboot#systemd-}"

    mkdir -p "$DESTDIR"/boot/efi/EFI/systemd "$DESTDIR"/boot/efi/EFI/BOOT
    cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR/boot/efi/EFI/systemd/$sdboot"
    cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR/boot/efi/EFI/BOOT/$fallback"

    mkdir -p "$DESTDIR"/efi/EFI/systemd "$DESTDIR"/efi/EFI/BOOT
    cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR/efi/EFI/systemd/$sdboot"
    cp "$DESTDIR"/usr/lib/systemd/boot/efi/systemd-bootx64.efi "$DESTDIR/efi/EFI/BOOT/$fallback"
done

```

Reply:
> Actually, maybe we can do away with this alltogether by writing a postinst script and running bootctl install. Let me try that out.

That wouldn't work on cross builds, I think. Assuming that's a possibility....

Reply:
Replaced with postinst script